test/infrastructure/docker/docker/kind_manager.go

```diff
@@ -51,25 +62,40 @@ func (m *Manager) CreateControlPlaneNode(name, image, clusterLabel, listenAddres
 		ListenAddress: listenAddress,
 		HostPort:      port,
 		ContainerPort: KubeadmContainerPort,
+		Protocol:      v1alpha4.PortMappingProtocolTCP,
 	})
-	node, err := createNode(
-		name, image, clusterLabel, constants.ControlPlaneNodeRoleValue, mounts, portMappingsWithAPIServer, ipFamily,
-		// publish selected port for the API server
-		append([]string{"--expose", fmt.Sprintf("%d", port)}, labelsAsArgs(labels)...)...,
-	)
+	createOpts := &nodeCreateOpts{
+		Name:         name,
+		Image:        image,
+		ClusterName:  clusterName,
+		Role:         constants.ControlPlaneNodeRoleValue,
+		PortMappings: portMappingsWithAPIServer,
+		Mounts:       mounts,
+		IPFamily:     ipFamily,
+	}
+	node, err := createNode(ctx, createOpts)
 	if err != nil {
 		return nil, err
 	}
 
 	return node, nil
 }
 
-func (m *Manager) CreateWorkerNode(name, image, clusterLabel string, mounts []v1alpha4.Mount, portMappings []v1alpha4.PortMapping, labels map[string]string, ipFamily clusterv1.ClusterIPFamily) (*types.Node, error) {
-	return createNode(name, image, clusterLabel, constants.WorkerNodeRoleValue, mounts, portMappings, ipFamily, labelsAsArgs(labels)...)
+func (m *Manager) CreateWorkerNode(ctx context.Context, name, image, clusterName string, mounts []v1alpha4.Mount, portMappings []v1alpha4.PortMapping, labels map[string]string, ipFamily clusterv1.ClusterIPFamily) (*types.Node, error) {
+	createOpts := &nodeCreateOpts{
+		Name:         name,
+		Image:        image,
+		ClusterName:  clusterName,
+		Role:         constants.WorkerNodeRoleValue,
+		PortMappings: portMappings,
+		Mounts:       mounts,
+		Labels:       labels,
+		IPFamily:     ipFamily,
+	}
+	return createNode(ctx, createOpts)
 }
 
-func (m *Manager) CreateExternalLoadBalancerNode(name, image, clusterLabel, listenAddress string, port int32, ipFamily clusterv1.ClusterIPFamily) (*types.Node, error) {
-	// gets a random host port for control-plane load balancer
+func (m *Manager) CreateExternalLoadBalancerNode(ctx context.Context, name, image, clusterName, listenAddress string, port int32, ipFamily clusterv1.ClusterIPFamily) (*types.Node, error) { // gets a random host port for control-plane load balancer
```

Comment:
nit: comment should probably be in the next line

Reply:
Weird, not sure what happened with these.